ballerina/tests/MedicalItemClient.bal

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerinax/mysql;
+import ballerinax/mysql.driver as _;
+import ballerina/sql;
+
+client class MedicalItemClient {
+
+    private final string entityName = "MedicalItems";
+    private final sql:ParameterizedQuery tableName = `MedicalItems`;
+    private final map<FieldMetadata> fieldMetadata = {
+        itemId: {columnName: "itemId", 'type: int},
+        name: {columnName: "name", 'type: string},
+        'type: {columnName: "type", 'type: string},
+        unit: {columnName: "unit", 'type: string}
+    };
+    private string[] keyFields = ["itemId"];
+
+    private SQLClient persistClient;
+
+    public function init() returns error? {
+        mysql:Client dbClient = check new (host = HOST, user = USER, password = PASSWORD, database = DATABASE, port = PORT);
+        self.persistClient = check new (self.entityName, self.tableName, self.fieldMetadata, self.keyFields, dbClient);
+    }
+
+    remote function create(MedicalItem value) returns int|error? {
+        sql:ExecutionResult result = check self.persistClient.runInsertQuery(value);
+
+        if result.lastInsertId is () {
+            return value.itemId;
+        }
+        return <int>result.lastInsertId;
+    }
+
+    remote function readByKey(int key) returns MedicalItem|error {
+        return (check self.persistClient.runReadByKeyQuery(key)).cloneWithType(MedicalItem);
+    }
+
+    remote function read(map<anydata>|FilterQuery? filter = ()) returns stream<MedicalItem, error?>|error {
```

Comment:
```suggestion
    remote function read(map<anydata>? filter = ()) returns stream<MedicalItem, error?>|error {
```

Reply:
remove the filter query part for the moment.  it is not okay to expose SQL syntax. We need to find another way

Reply:
Fixed in [7600af1](https://github.com/ballerina-platform/module-ballerina-persist/pull/1/commits/7600af1975f3074d3823f5aa3989289aba4dfe3e)